File: domain/src/main/java/com/crhonvas/domain/model/comments/Comment.java
package com.crhonvas.domain.model.comments;

import androidx.annotation.NonNull;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class Comment {

    private Integer postId;
    private Integer id;
    private String name;
    private String email;
    private String body;

    private Comment(@NonNull Builder builder) {
        this.postId = builder.postId;
        this.id = builder.id;
        this.name = builder.name;
        this.email = builder.email;
        this.body = builder.body;
    }

    @NonNull
    public Integer getPostId() {
        return postId == null ? -1 : postId;
    }

    @NonNull
    public Integer getId() {
        return id == null ? -1 : id;
    }

    @NonNull
    public String getName() {
        return name == null ? "" : name;
    }

    @NonNull
    public String getEmail() {
        return email == null ? "" : email;
    }

    @NonNull
    public String getBody() {
        return body == null ? "" : body;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass()) return false;

        Comment that = (Comment) o;

        return new EqualsBuilder()
                .append(postId, that.postId)
                .append(id, that.id)
                .append(name, that.name)
                .append(email, that.email)
                .append(body, that.body)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37)
                .append(postId)
                .append(id)
                .append(name)
                .append(email)
                .append(body)
                .toHashCode();
    }

    public static final class Builder {
        private Integer postId = -1;
        private Integer id = -1;
        private String name = "";
        private String email = "";
        private String body = "";

        public Builder() {}

        @NonNull
        public Builder postId(@NonNull Integer postId) {
            this.postId = postId;
            return this;
        }

        @NonNull
        public Builder id(@NonNull Integer id) {
            this.id = id;
            return this;
        }

        @NonNull
        public Builder name(@NonNull String name) {
            this.name = name;
            return this;
        }

        @NonNull
        public Builder email(@NonNull String email) {
            this.email = email;
            return this;
        }

        @NonNull
        public Builder body(@NonNull String body) {
            this.body = body;
            return this;
        }

        @NonNull
        public Comment build() {
            return new Comment(this);
        }
    }
}

File: domain/src/main/java/com/crhonvas/domain/model/comments/Comment.kt
package com.crhonvas.domain.model.comments

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder

class Comment private constructor(builder: Builder) {
    private val postId: Int?
    private val id: Int?
    private val name: String?
    private val email: String?
    private val body: String?

    init {
        postId = builder.postId
        id = builder.id
        name = builder.name
        email = builder.email
        body = builder.body
    }

    fun getPostId(): Int {
        return postId ?: -1
    }

    fun getId(): Int {
        return id ?: -1
    }

    fun getName(): String {
        return name ?: ""
    }

    fun getEmail(): String {
        return email ?: ""
    }

    fun getBody(): String {
        return body ?: ""
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as Comment
        return EqualsBuilder()
            .append(postId, that.postId)
            .append(id, that.id)
            .append(name, that.name)
            .append(email, that.email)
            .append(body, that.body)
            .isEquals
    }

    override fun hashCode(): Int {
        return HashCodeBuilder(17, 37)
            .append(postId)
            .append(id)
            .append(name)
            .append(email)
            .append(body)
            .toHashCode()
    }

    class Builder {
        var postId = -1
        var id = -1
        var name = ""
        var email = ""
        var body = ""
        fun postId(postId: Int): Builder {
            this.postId = postId
            return this
        }

        fun id(id: Int): Builder {
            this.id = id
            return this
        }

        fun name(name: String): Builder {
            this.name = name
            return this
        }

        fun email(email: String): Builder {
            this.email = email
            return this
        }

        fun body(body: String): Builder {
            this.body = body
            return this
        }

        fun build(): Comment {
            return Comment(this)
        }
    }
}